unrarall

```diff
@@ -170,6 +179,36 @@ function getUnrarFlags()
   esac
 }
 
+#function to check if an archive has been extracted already
+function isAlreadyExtracted()
+{
+  case "$1" in
+    unrar)
+      file_listing_cmd="unrar lb \"$2\""
+    ;;
+    7z)
+      file_listing_cmd="7z l \"$2\" | sed -n -e '/-------------------/,/-------------------/ p' | sed '1d;\$d' | awk '{out=\$6; for(i=7;i<=NF;i++){out=out\" \"\$i}; print out}'"
```

Comment:
*shakes fist at Igor Pavlov*

Reply:
So I got this down to:
```7z L -ba -slt \"$2\" | grep Path | cut -d '=' -f 2```
but it only works with recent 7z versions.

Reply:
That looks a lot less awful and a lot more maintainable.